network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py

```diff
@@ -0,0 +1,131 @@
+
+import json
+
+from django.db import models
+from modelcluster.fields import ParentalKey
+from wagtail.admin.edit_handlers import FieldPanel, InlinePanel, StreamFieldPanel, MultiFieldPanel
+from wagtail.core.blocks import RichTextBlock
+from wagtail.core.fields import StreamField
+from wagtail.core.models import TranslatableMixin, Page, Orderable
+from wagtail_localize.fields import SynchronizedField, TranslatableField
+from wagtail.snippets.edit_handlers import SnippetChooserPanel
+
+from ..customblocks.full_content_rich_text_options import full_content_rich_text_options
+from networkapi.wagtailpages.pagemodels import customblocks
+from networkapi.wagtailpages.pagemodels.buyersguide.utils import get_categories_for_locale
+from networkapi.wagtailpages.pagemodels.mixin.foundation_metadata import FoundationMetadataPageMixin
+from networkapi.wagtailpages.utils import get_language_from_request
+
+
+class BuyersGuideCampaignPageDonationModalRelation(TranslatableMixin, Orderable):
+    page = ParentalKey(
+        'BuyersGuideCampaignPage',
+        related_name='donation_modal_relations',
+    )
+    donation_modal = models.ForeignKey(
+        'DonationModal',
+        null=False,
+        blank=False,
+        on_delete=models.CASCADE,
+        help_text='Choose existing or create new donation modal'
+    )
+
+    panels = [
+        SnippetChooserPanel('donation_modal'),
+    ]
+
+    class Meta(TranslatableMixin.Meta, Orderable.Meta):
+        verbose_name = 'Donation Modals'
+        verbose_name_plural = 'Donation Modals'
+
+
+class BuyersGuideCampaignPage(FoundationMetadataPageMixin, Page):
+    parent_page_types = ['BuyersGuideEditorialContentIndexPage']
+    subpage_types: list = []
+    template = 'pages/buyersguide/campaign_page.html'
+
+    cta = models.ForeignKey(
+        'CTA',
+        related_name='buyersguide_campaign_page_for_cta',
```

Comment:
Can you add some context in what case the backward relation would be used?

Reply:
The forward relation makes sense of course, from the page to the CTA. But do we ever need to go from the CTA to a page?